src/main/java/com/iota/iri/network/Node.java

```diff
@@ -265,6 +267,14 @@ public void preProcessReceivedData(byte[] receivedData, SocketAddress senderAddr
 
                 } catch (NoSuchAlgorithmException e) {
                     log.error("MessageDigest: " + e);
+                } catch (final TransactionValidator.StaleTimestampException e) {
+                    log.error(e.getMessage());
+                    try {
+                        transactionRequester.clearTransactionRequest(receivedTransactionHash);
+                    } catch (Exception e1) {
+                        log.error(e1.getMessage());
+                    }
+                    neighbor.incInvalidTransactions();
```

Comment:
put in `finally` block maybe?

Reply:
this should only happen if an exception is thrown, not always.